src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -0,0 +1,583 @@
+    azure_arc_agent_version = get_agent_version(configuration)
+
+    # Create connected cluster resource object
+    aad_profile = ConnectedClusterAADProfile(
+        tenant_id="",
+        client_app_id="",
+        server_app_id=""
+    )
+    identity = ConnectedClusterIdentity(
+        type="SystemAssigned"
+    )
+    if tags is None:
+        tags = {}
+    cc = ConnectedCluster(
+        location=location,
+        identity=identity,
+        agent_public_key_certificate=public_key,
+        aad_profile=aad_profile,
+        kubernetes_version=kubernetes_version,
+        total_node_count=total_node_count,
+        agent_version=azure_arc_agent_version,
+        tags=tags
+    )
+    return cc
+
+
+def get_pod_dict(api_instance):
+    pod_dict = {}
+    timeout = time.time() + 60
+    while not pod_dict:
+        try:
+            api_response = api_instance.list_namespaced_pod('azure-arc')
+            for pod in api_response.items:
+                pod_dict[pod.metadata.name] = 0
+            return pod_dict
+        except Exception as e:  # pylint: disable=broad-except
+            logger.warning("Error occurred when retrieving pod information: %s", e)
+            time.sleep(5)
+        if time.time() > timeout:
+            logger.warning("Unable to fetch azure-arc agent pods.")
+            return pod_dict
+
+
+def check_pod_status(pod_dict):
+    v1 = kube_client.CoreV1Api()
+    w = watch.Watch()
+    for event in w.stream(v1.list_namespaced_pod, namespace='azure-arc', timeout_seconds=360):
+        pod_status = event['raw_object'].get('status')
+        pod_name = event['object'].metadata.name
+        if pod_status.get('containerStatuses'):
+            for container in pod_status.get('containerStatuses'):
+                if container.get('name') in pod_name:
+                    if container.get('state').get('running') is None:
+                        pod_dict[pod_name] = 0
+                    else:
+                        pod_dict[pod_name] = 1
+                    if container.get('state').get('terminated') is not None:
+                        logger.warning("%s%s%s", "The pod {} was terminated. ".format(container.get('name')),
+                                       "Please ensure it is in running state once the operation completes. ",
+                                       "Run 'kubectl get pods -n azure-arc' to check the pod status.")
+        if all(ele == 1 for ele in list(pod_dict.values())):
+            return
+    logger.warning("%s%s", 'The pods were unable to start before timeout. ',
+                   'Please run "kubectl get pods -n azure-arc" to ensure if the pods are in running state.')
+
+
+def get_connectedk8s(cmd, client, resource_group_name, cluster_name):
+    return client.get(resource_group_name, cluster_name)
+
+
+def list_connectedk8s(cmd, client, resource_group_name=None):
+    if resource_group_name is None:
+        return client.list_by_subscription()
```

Comment:
Please use `if not resource_group_name` as we allow the workaround of passing an empty resource group `-g=""` to temporarily disable configured default resource group in list command to list by subscription.